### PR TITLE
Fixes mineral turfs having weird lighting

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Pixel offsets, unlike transforms, offset overlays too. this was breaking lighting overlays for mineral walls.

We did pixel offsets to save on init time, but we can acomplish the same thing using an initial matrix. It's static, so there's no additional cost. S free

Damn moth

## Changelog
:cl:
fix: Mining walls won't have fucked lighting anymore
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
dont damn moth me im the one who found the transform type var optimization

File: code/__DEFINES/matrices.dm [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Helper macro for compile time translation
```
Comment (Mothblocks) [inline code/__DEFINES/matrices.dm:1]:
```suggestion
/// Helper macro for compile time translation
```

File: code/game/turfs/closed/minerals.dm [hunk 2]
```diff
@@ -17,10 +15,10 @@
...
 	base_icon_state = "smoothrocks"
 
-	base_pixel_x = MINERAL_WALL_OFFSET
-	base_pixel_y = MINERAL_WALL_OFFSET
-	pixel_x = MAP_SWITCH(MINERAL_WALL_OFFSET, 0)
-	pixel_y = MAP_SWITCH(MINERAL_WALL_OFFSET, 0)
+	// This is static
+	// Done like this to avoid needing to make it dynamic and save cpu time
+	// 4 to the left, 4 down
+	transform = TRANSLATE_MATRIX(-4, -4)
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:21]: Does this show up in the map editor (BYOND's or otherwise)? pixel_x and pixel_y were making this look bad

**Comment by LemonInTheDark** [comment]
> dont damn moth me im the one who found the transform type var optimization

you told me to fix it because you don't do git, it's grunt work

File: code/game/turfs/closed/minerals.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/turfs/closed/minerals.dm:21]: it will show up in 515 dm, I can use the compile time macro if you'd like

**Review by Mothblocks — APPROVED** [review]
correct

**Pull request merged**